src/pages/Proposal.vue

```diff
@@ -144,9 +151,116 @@ export default defineComponent({
 			ProposalType,
 		};
 	},
+	setup() {
+		const formState = reactive(ProposalStore());
+		const walletStore = reactive(WalletStore());
+		const daoStore = reactive(DaoID());
+
+		return {
+			formState,
+			walletStore,
+			daoStore,
+			validateMessages: VALIDATE_MESSAGES,
+		};
+	},
 	methods: {
-		onFinish(values: Event) {
-			console.log("Success:", values);
+		async onFinish(values: any) {
+			try {
+				let {
+					amount,
+					recipient,
+					url,
+					url_hash,
+					proposal_type,
+					vote_date,
+					message,
+					asaId,
+				} = values;
+				if (typeof this.daoStore.dao_id === "undefined") {
+					console.error("appId not defined");
+					return;
+				}
+				if (typeof this.daoStore.govt_id === "undefined") {
+					console.error("govt_id not defined");
+					return;
+				}
+				let lsig: LogicSigAccount = await getProposalLsig(
+					this.daoStore.dao_id,
+					this.walletStore.address
+				);
+				let daoLsig: LogicSigAccount = await getDaoFundLSig(
+					this.daoStore.dao_id
+				);
+				const startTime = convertToSeconds(vote_date[0]);
+				const endTime = convertToSeconds(vote_date[1]);
+				const executeBefore = endTime + 7 * 60; // end time + 7 minutes in seconds
```

Comment:
Let create one more filed `executeBefore` in add proposal form.

Reply:
@robert-zaremba Please confirm if this field also has to be added in form.